trillian_map_api.proto

```diff
@@ -152,6 +165,8 @@ service TrillianMap {
   rpc GetLeaves(GetMapLeavesRequest) returns (GetMapLeavesResponse) {}
   rpc GetLeavesByRevision(GetMapLeavesByRevisionRequest)
       returns (GetMapLeavesResponse) {}
+  // GetLastInRangeByRevision returns the last leaf in a requested range.
```

Comment:
What do you mean by "last"?   Bear in mind that the API definition still includes a statement that "`Map entries do not have a well defined ordering`".

Reply:
This is perhaps an item for discussion:

Classical computer science Map data structures do not have a stable ordering for their keys, and Golang goes out of it's way to remind developers of that.  Was this the idea behind the API statement?

This merkle tree data structure is a prefix tree. Elements are placed in the tree by following a path, left (0) and right (1) down from the root.   By definition its elements do have an ordering, and the merkle tree data structure commits to that ordering. 

Putting a "future proofing" hat on, all the cryptographic commitments to key/value pairs that I am aware of also commit to ordering. The one exception to this is the homomorphic hashing proposal [1] who's purpose is to implement combinable signatures over sets. 

[1] https://code.fb.com/security/homomorphic-hashing/

Reply:
tl;dr - prob'ly best just to remove the ordering comment.